server/rest.py

```diff
@@ -22,351 +22,127 @@
-
-    :param conn: the database connector.  Used for validating fields.
-    :param fields: a list of known fields.
-    :param filter: either a dictionary or a list or tuple with two to three
-                   components representing (field), [(operator),] (value).
-    :returns filter: the filter in dictionary-style.
-    """
-    if isinstance(filter, (list, tuple)):
-        if len(filter) < 2 or len(filter) > 3:
-            raise RestException('Filters in list format must have two or '
-                                'three components.')
-        if len(filter) == 2:
-            filter = {'field': filter[0], 'value': filter[1]}
-        else:
-            filter = {
-                'field': filter[0], 'operator': filter[1], 'value': filter[2]
-            }
-    if filter.get('operator') not in dbs.FilterOperators:
-        raise RestException('Unknown filter operator %r' % filter.get(
-            'operator'))
-    filter['operator'] = dbs.FilterOperators[filter.get('operator')]
-    if 'field' not in filter and 'lvalue' in filter:
-        filter['field'] = {'value': filter['lvalue']}
-    if 'field' not in filter and ('func' in filter or 'lfunc' in filter):
-        filter['field'] = {
-            'func': filter.get('func', filter.get('lfunc')),
-            'param': filter.get('param', filter.get('params', filter.get(
-                'lparam', filter.get('lparams'))))
-        }
-    if 'value' not in filter and 'rfunc' in filter:
-        filter['value'] = {
-            'func': filter['rfunc'],
-            'param': filter.get('rparam', filter.get('rparams'))
-        }
-    if 'field' not in filter:
-        raise RestException('Filter must specify a field or func.')
-    if (not conn.isField(
-            filter['field'], fields,
-            allowFunc=getattr(conn, 'allowFilterFunctions', False)) and
-            not isinstance(filter['field'], dict) and
-            'value' not in filter['field'] and 'func' not in filter['field']):
-        raise RestException('Filters must be on known fields.')
-    if not filter.get('value'):
-        raise RestException('Filters must have a value or rfunc.')
-    if not conn.checkOperatorDatatype(filter['field'], filter['operator'],
-                                      fields):
-        raise RestException('Cannot use %s operator on field %s' % (
-            filter['operator'], filter['field']))
-    return filter
+from . import assetstore, dbs
+from .assetstore import dbInfoKey
+from .query import DatabaseQueryException, dbFormatList, queryDatabase, \
+    preferredFormat
 
 
-class DatabaseItemResource(Item):
+class DatabaseFileResource(File):
 
     def __init__(self, apiRoot):
-        # Don't call the parent (Item) constructor, to avoid redefining routes,
+        # Don't call the parent (File) constructor, to avoid redefining routes,
         # but do call the grandparent (Resource) constructor
-        super(Item, self).__init__()
+        super(File, self).__init__()
```

Comment:
Maybe instead of having this descend from `File`, we just pass `info['apiRoot'].file` into it and have it tack stuff on? Then we don't have to deal with the constructor messiness.

```
class DatabaseFileResource(ModelImporter):
    def __init__(self, fileResource):
        fileResource.route(...)
```